tests/interfaces/records/handlers/records-read.spec.ts

```diff
@@ -135,6 +138,189 @@ describe('RecordsReadHandler.handle()', () => {
+          requester: caroline,
+          protocol,
+          protocolDefinition
+        });
+        const protocolWriteReply = await dwn.processMessage(caroline.did, protocolsConfig.message, protocolsConfig.dataStream);
+        expect(protocolWriteReply.status.code).to.equal(202);
+
+        // Alice writes image to Caroline's DWN with Bob as recipient
+        const encodedImage = new TextEncoder().encode('cafe-aesthetic.jpg');
+        const imageRecordsWrite = await TestDataGenerator.generateRecordsWrite({
+          requester    : alice,
+          protocol,
+          schema       : socialMediaProtocolDefinition.labels.image.schema,
+          data         : encodedImage,
+          recipientDid : bob.did
+        });
+        const imageReply = await dwn.processMessage(caroline.did, imageRecordsWrite.message, imageRecordsWrite.dataStream);
+        expect(imageReply.status.code).to.equal(202);
+
+        // Alice writes caption to her image to Caroline's DWN with Bob as recipient
+        const imageContextId = await imageRecordsWrite.recordsWrite.getEntryId();
+        const encodedCaption = new TextEncoder().encode('coffee and work vibes!');
+        const captionRecordsWrite = await TestDataGenerator.generateRecordsWrite({
+          requester : alice,
+          protocol,
+          schema    : socialMediaProtocolDefinition.labels.caption.schema,
+          contextId : imageContextId,
+          parentId  : imageContextId,
+          data      : encodedCaption
+        });
+        const captionResponse = await dwn.processMessage(caroline.did, captionRecordsWrite.message, captionRecordsWrite.dataStream);
+        expect(captionResponse.status.code).to.equal(202);
+
+        // Bob (recipient) reads Alice's caption
+        const bobCaptionRecordsRead = await RecordsRead.create({
+          recordId                    : captionRecordsWrite.message.recordId,
+          authorizationSignatureInput : Jws.createSignatureInput(bob)
+        });
+        const bobCaptionReadReply = await dwn.processMessage(caroline.did, bobCaptionRecordsRead.message);
+        expect(bobCaptionReadReply.status.code).to.equal(200);
+
+        // Dave (anyone) is not able to read Alice's caption
+        const daveCaptionRecordsRead = await RecordsRead.create({
+          recordId                    : captionRecordsWrite.message.recordId,
+          authorizationSignatureInput : Jws.createSignatureInput(dave)
+        });
+        const daveCaptionReadReply = await dwn.processMessage(caroline.did, daveCaptionRecordsRead.message);
+        expect(daveCaptionReadReply.status.code).to.equal(401);
+        expect(daveCaptionReadReply.status.detail).to.include('inbound message action \'read\' not in list of allowed actions');
+      });
+
+      // also test blocking of non-author
+      it('should allow read with author rule', async () => {
+        // scenario: Alice writes an image to Caroline's DWN, then Bob writes a reply to the image.
```

Comment:
Similar to earlier comment: not sure if Dave not allowed to read Bob's reply/comment is common? Consider an easier-to-understand scenario so a code reader/reviewer doesn't have to inspect the protocol definition to confirm understanding. Maybe simple messaging between alice and bob?

Reply:
Refactored tests to use the email protocol when testing author and recipient reads. For the anyone test, I'm still using the socialMedia protocol in a simplified scenario since it doesn't make sense to have allow anyone to read emails.